### ISSUE TITLE
Using notifier.notify without component nvim

### ISSUE THREAD

**Issue by folke** [vigoux/notifier.nvim#19]
https://github.com/vigoux/notifier.nvim/blob/89e7d10aa9db1e9e55785069a4aa69982d32fcca/lua/notifier/status.lua#L201

If you don't configure any components, message sent through notifier.notify will never be displayed.

Reason I dont want to confiure `vim.notify`, is because **Noice** itself can also be `vim.notify`, which makes it easy to route any messages to different views, like notify or notifier based on filters.

I'm willing to make a PR to make this works, but not sure what the best way would be. Any suggestions?

**Comment by vigoux** [comment]
Yeah this is indeed a limitation.
I think that the only thing to do is check if the `nvim` component is displayed _before_ overriding `vim.notify`.

You can make a PR for that if you want, I'd be glad to include it !

**Comment by vigoux** [comment]
Okay, looking back at this issue, I think that you should rather use the `status.push` and `status.pop` interface.

This should be the most configurable way of displaying things.

**Comment by dlvhdr** [comment]
Just wanted to say this plugin looks really nice, and I would love to use it instead of nvim-notify together with Noice 💖

**Comment by megalithic** [comment]
👆 agreed; both plugins are great for notifications, but man i love the simplicity of notifier

**Comment by folke** [comment]
@vigoux you'll have to tell me exactly how you would like this to work :) I dont fully understand your comment.

By exposing `status.push`, the limitation of configured components still applies (see my PR)

Unless I would add a custom component like `noice`.

But then again, all messages would be shown as `noice`, which is not right. In Noice, we route messages from multiple providers to destinations. Providers can be `vim.notify`, `Neovim messages`, ...

Cache and lifetime is also handled by the `notify` function in `notifier` which is great. I don't think it makes a lot of sense to re-implement all of that in external plugins.

Ideally, I would want to have an exposed interface like `vim.notify` (as is currently the case), where I can tell notifier to show the message.

**Comment by vigoux** [comment]
Random pile of thoughts that partially answer the question:

For now the `components` configuration key is used for two reasons:
- They configure _which_ component to display (allow to disable some components)
- They configure the _order_ in which to configure it

That means that (to me) changing the configuration (or the behavior) is not actually the good path.

I am not sure _why_ this is needed though: if the user wants to use notifier, then there's probably some configuration to do anyway.

**Comment by folke** [comment]
Just to clarify, Noice is basically a message router. It routes messages from sources to destintations using routes.

## Sources
- `ext_messages`: any Neovim message through `vim.ui_attach` with `ext_messages`
- `vim.notify`: routes any messages from `vim.notify`
- `lsp_progress`
- ...

## Destinations (Views)
- popup
- split
- vsplit
- **nvim-notify**
- virtualtext
- mini: builtin view similar to **notifier.nvim**

## Example Routes
Using Noice, users can for example:
- send all lsp progress info to `mini`
- send all messages with more than 10 lines to `split`
- send any errro / warning to `nvim-notify`
- show any info/debug/trace `vim.notify` message in `mini`
- show any `echo` or `echom` in `mini`
- ...

## Integration
What Noice users would like to see in terms of an integration, is adding `notifier.nvim` as a **destination** (**view**).

Basically, in all the examples, replace `mini` by `notifier.nvim` :)

As you can also see, Noice users probably want Noice to handle `vim.notify` and possibly `lsp_progress`. Depending on the message, it can still be forwarded to `notifier.nvim` or another destination.

Ideally, users should be able to specify a component in their route options. Which would typically coresspond with some kind of event source.

I can use the status API for all this (or the notify api), but it will never work as long as https://github.com/vigoux/notifier.nvim/pull/20/files is not merged.

Unless you see another way?

**Comment by folke** [comment]
I can actually make it work without my PR, but that doesn't feel right?.

```lua
require("notifier").setup({ components = {} })
require("notifier.config").config.components = {"nvim", "lsp"}
```

From within Noice, I can then simply update the `require("notifier.config").config.components` when needed.

**Comment by vigoux** [comment]
My only question is _why_?
Why not just setting up the components right away?

**Comment by folke** [comment]
@vigoux because I don't want notifier to hijack `vim.notify` as I explained before? Noice is far more powerful to route messages to different destinations, so I prefer noice to handle `vim.notify`.

But i would still like to route some messages coming from `vim.notify` or another source to notifier.

I'm personally more than happy with the noice mini view, and was just planning to add notifier for your existing users, so feel free to close this issue, if you feel this doesn't make a lot of sense.

**Comment by vigoux** [comment]
My point is: the PR you did earlier to expose the `notify` function already solved that use case ?

The question of components (which are to be thought as some kind of "UI elements" for `notifier`), is _not related_ to the use of the notify function from what I see here.

For a more flexible interface, you should be able to get away with using the `status.push`, `status.pop` API (as already noted in #22).

I'll indeed close this issue as it seems fixed to (and to you it seems ?).

Regarding `noice` in itself: do whatever you want ! Plus I am not in a "user war" to if people want to use `noice` instead of this plugin, I am 100% fine with it.

**Comment by folke** [comment]
> I'll indeed close this issue as it seems fixed to (and to you it seems ?).

Noted

> Regarding `noice` in itself: do whatever you want ! Plus I am not in a "user war" to if people want to use `noice` instead of this plugin, I am 100% fine with it.

No worries, neither am I. I'm having a blast creating plugins. Lots of fun. That's what I'm doing it for. Thanks for all your hard work on Neovim!

**Comment by megalithic** [comment]
So, at the moment, noice.nvim and notifier.nvim are not/will not be compatible is what I'm seeing out of this closed issue? Just confirming for others that may come here and want a TL;DR.

**Comment by dlvhdr** [comment]
Yes, but the `mini` view from noice looks very good, so I'm using it
